### ISSUE TITLE
State of replica after itouch to S3 is 'X'

### ISSUE THREAD

**Issue by JustinKyleJames** [irods/irods_resource_plugin_s3#1978]
The state of the replica after itouch to S3 is 'X' and the data object remains locked.

The object is created successfully in S3 and there are no S3 plugin type errors generated.  (Also zero length puts work.)  The following error is in the logs.

ERROR: [finalize_replica_opened_for_create_or_write:75] - no entry found in replica state table for [/tempZone/home/rods/zerofile]

The replica status is good after itouch to a unixfilesystem resource so this has something to do with the interplay between irods and the S3 plugin.

**Comment by trel** [comment]
option 2 might also cause other additional policy to fire in S3 or downstream for other tooling watching for any S3 events.

seems option 1 would be clean enough to implement - it's always going to be zero after a create... right?

**Comment by alanking** [comment]
Should we transfer this to irods/irods? irods/irods#6479 duplicates this issue and the fix is in the server, in my opinion. If `replica_close` calls the close resource plugin operation before attempting to finalize the data object (i.e. stat), the issue is resolved without any further changes.

The changes proposed above (https://github.com/irods/irods_resource_plugin_s3/issues/1978#issuecomment-857075611) may well be things that should be implemented, but I don't know much about S3 expectations. Thoughts, @JustinKyleJames / @korydraughn / @trel?

**Comment by korydraughn** [comment]
Sounds reasonable. Have you already confirmed that this issue goes away if `replica_close` is tweaked?

**Comment by alanking** [comment]
Yes, I have confirmed that `itouch` is now working as expected with the S3 resource with the `replica_close` changes.

**Comment by korydraughn** [comment]
Good. Proceed with transferring this issue to the irods/irods repo.
